File: Cplusplus/로또의최고순위와최저순위.cpp
//22.01.04 로또의 최고 순위와 최저 순위 by CJHofficial

#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;

    int bestRank = 7;
    int worstRank = 7;

    for (int i : win_nums) {
        for (int j : lottos) {
            if (i == j) worstRank -= 1;
        }
    }
    bestRank = worstRank;
    for (int j : lottos) {
        if (j == 0) bestRank -= 1;
    }

    if (bestRank > 5) bestRank = 6;
    if (worstRank > 5) worstRank = 6;
    answer.push_back(bestRank);
    answer.push_back(worstRank);

    return answer;
}

File: Python/level1/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    erasedNumberCount = countErasedNumber(lottos)
    minRanking = countSameNumber(lottos,win_nums)
    maxRanking = minRanking + erasedNumberCount
    answer = [changeRanking(maxRanking), changeRanking(minRanking)]
    return answer

def countErasedNumber(lottos):
    count = 0
    for lottoNumber in lottos:
        if lottoNumber == 0:
            count = count + 1
    return count

def countSameNumber(lottos,win_nums):
    count = 0
    for lotto in lottos:
        if lotto in win_nums:
            count = count + 1
    return count

def changeRanking(sameNumber):
    ranking = 6
    if sameNumber == 2:
        ranking=5
    elif sameNumber == 3:
        ranking=4
    elif sameNumber == 4:
        ranking=3
    elif sameNumber == 5:
        ranking=2
    elif sameNumber == 6:
        ranking=1
    return ranking
